syncer/syncer.go

```diff
@@ -497,9 +497,13 @@ func (s *Syncer) peerLoop(closeChan <-chan struct{}) error {
 			if numOutbound() >= s.config.MaxOutboundPeers || closing() {
 				break
 			}
+
 			// NOTE: we don't bother logging failure here, since it's common and
 			// not particularly interesting or actionable
-			if _, err := s.Connect(p); err == nil {
+			//
+			// NOTE: we can safely pass the background context here since the
+			// connect timeout will be enforced
+			if _, err := s.Connect(context.Background(), p); err == nil {
```

Comment:
Should `ctx, cancel := context.WithTimeout(context.Background(), s.config.ConnectTimeout)` be moved out of `Connect()` to give users full control over the lifecycle when calling `Connect()` directly?

Reply:
yes